builder/openstack/builder.go

```diff
@@ -62,6 +62,11 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 		return nil, fmt.Errorf("Error initializing compute client: %s", err)
 	}
 
+	// By default, instance name is same as image name
+	if b.config.InstanceName == "" {
+		b.config.InstanceName = b.config.ImageName
+	}
+
```

Comment:
This should move up to `Prepare`.

Reply:
I really did not want to like this because I do feel `Prepare` is the right place to input validation and at first I actually tried. But I could not help it because `RunConfig struct` in `run_config.go` does not have ImageName variable in `ImageConfig` in `ImageConfig struct'.

So I just choose the place at least without big change, but maybe it's not the right one I felt. I'm glad to fix up if you tell me the right way. Thanks.

Reply:
You should be able to move this code up to here and have it work like you expect https://github.com/hashicorp/packer/pull/6041/files#diff-0062cd14734ce9ce02373f2154b690e0R54

When validating data that spans multiple configuration structs, I think it makes sense to do it at the outer level, in the builder's prepare.

Reply:
Oh, I did not notice the `Builder's prepare`. It makes sense, and I moved. Thanks for kind guide!